### PR TITLE
Simplify repository

### PR DESCRIPTION
**Pull request by codeliner**
- Stream strategies are removed and stream selection is now handled by the repository
- Aggregate roots are only kept in memory until pending events are add to the stream
- Apply events late feature is reverted as it is no longer needed
- AggregateRepositoryFactory is aligned

### CODE REVIEW

File: tests/Mock/DefaultAggregateRoot.php [hunk 1]
```diff
@@ -71,9 +71,9 @@ public function popRecordedEvents()
     }
 
     /**
-     * @param Message $event
+     * @param $event
      */
-    public function apply(Message $event)
+    public function apply($event)
```
Comment (prolic) [inline tests/Mock/DefaultAggregateRoot.php]: We still rely on the Message interface, no?

Comment (codeliner) [inline tests/Mock/DefaultAggregateRoot.php]: Was clarified in the chat. But I'll add a short note for history:
We rely on Message interface on the event store side but the mock is used to simulate an aggregate of domain which must not rely on the Message interface as an AggregateTranslator can translate custom domain events into Message compatible events

**Comment by prolic** [comment]
wip can be removed?

**Comment by codeliner** [comment]
yes, wip is removed. Ready for final review

File: docs/repositories.md [hunk 2]
```diff
@@ -85,28 +102,37 @@ We can identify 6 dependencies but all are optional.
...
+If you wish to use another name, you can pass a custom [StreamName](../src/Stream/StreamName.php) to the repository.
+This is especially useful when you want to have an event stream per aggregate type, for example store all user related events
+in a `user_stream`.
 
-Now that you know the customization options you may ask: **How to put all that together?**
-The answer is: **With a repository!**
-The best way to see it in action is by looking at the [AggregateRepositoryTest](../tests/Aggregate/AggregateRepositoryTest.php).
+The repository can also be configured to create a new stream for each new aggregate instance. You need to turn the last
+constructor parameter `oneStreamPerAggregate` to true to enable the mode.
+This can be useful when working for example with mongoDb and you want to persist all events of an aggregate in single document (take care of the document size limit).
```
Comment (prolic) [inline docs/repositories.md]: mongoDb => MongoDB

**Comment by codeliner** [comment]
@prolic mongoDB corrected

**Pull request merged**